### DISCUSSION TITLE
Question: How often do you use multiple queries to synchronize a single table?

### DISCUSSION THREAD

**Discussion by adrianhall** [CommunityToolkit/Datasync#71] [General]
Hey folks,

In the new `OfflineDbContext`, I'm trying to streamline the experience.  Right now, the code I have requires you to set up on `OnDatasyncInitialization()` builder:

```csharp
public class MyDbContext : OfflineDbContext
{
  protected override void OnDatasyncInitialization(DatasyncOfflineOptionsBuilder builder)
  {
    builder.SetHttpClientFactory(new HttpClientFactory("https://datasync.foo.com"));
    builder.Entity<MyEntity>(options => 
    {
      options.Endpoint = new Uri("/tables/myentity", UriKind.Relative);
      options.Query.Where(x => x.MyProperty == 1);
    });
  }
}
```

My question is for your use cases.  Under what circumstances do you pull through multiple queries.  For instance, in the old code base, you could use service.PullAsync(query, ...); - how often do you do this to pull multiple times for the same table?

**Comment by hansmbakker** [comment]
We don't do filtering client-side but server-side, so on the client-side we do `Pullasync` once per table without a specific query.

**Reply by powerdude** [reply]
Since I have a lot of parent-child relationships, I tend to pull children based on the parent id quite often since it's a tiny bit faster than trying to always pull all the children.

**Reply by adrianhall** [reply]
Thanks @powerdude - so do you pull the parent table, then iterate over the parent table IDs to do individual pulls with a query and unique query ID?

**Reply by powerdude** [reply]
yes, as the user navigates through the rows/screens.
